department-of-veterans-affairs/caseflow
# Add roles and ARIA attributes to components in Searchable Dropdown

Resolves [CASEFLOW-1017](https://vajira.max.gov/browse/CASEFLOW-1017)

### Description
Add roles and aria attributes to components in SearchableDropdown for accessibility. 
By default, the components in React Select [aren't supplied aria attributes or roles ](https://github.com/JedWatson/react-select/pull/4322).

[Relevant ARIA documentation on combobox](https://www.w3.org/TR/wai-aria-1.1/#combobox)
[Setting up JAWS in VM on MacOS](https://github.com/department-of-veterans-affairs/appeals-team/blob/master/Engineering/accessibility-jaws-setup.md)

### Acceptance Criteria
- [ ] Screenreader reads input as combobox

### Testing Plan
1. Start up virtual machine with JAWS [(instructions)](https://github.com/department-of-veterans-affairs/appeals-team/blob/master/Engineering/accessibility-jaws-setup.md)
2. Start Caseflow application locally. Be sure to change where webpack bundle is served from in head.html.erb as noted in instructions 
2. Switch to intake user 
3. Go to `intake`
4. Navigate to the `Which form are you processing` dropdown
5. Verify that the element is read as a listbox instead of an editable text field


https://user-images.githubusercontent.com/68879427/118894254-be3b9080-b8b8-11eb-886c-c11622b010d1.mov
